client/app/bundles/course/assessment/question/programming/actions/onlineEditorActionCreators.jsx

```diff
@@ -33,25 +33,81 @@ export function deletePythonTestCase(testType, index) {
   };
 }
 
-export function updateNewDataFile(filename, index) {
+export function updatePythonNewDataFile(filename, index) {
   return {
     type: actionTypes.PYTHON_NEW_DATA_FILE_UPDATE,
     index,
     filename,
   };
 }
 
-export function deleteNewDataFile(index) {
+export function deletePythonNewDataFile(index) {
   return {
     type: actionTypes.PYTHON_NEW_DATA_FILE_DELETE,
     index,
   };
 }
 
-export function deleteExistingDataFile(filename, toDelete) {
+export function deletePythonExistingDataFile(filename, toDelete) {
   return {
     type: actionTypes.PYTHON_EXISTING_DATA_FILE_DELETE,
     filename,
     toDelete,
   };
 }
+
+export function updateCppCodeBlock(field, newValue) {
```

Comment:
I'm not sure if you can find other ways of declaring the ActionCreators, because this method results in the number of actions scaling proportionately with the number of languages. I looked through some [online discussions trying to discuss this](https://github.com/reactjs/redux/issues/786) but there is no 'correct answer'.

In my head, what I was thinking of is since `updateCodeBlock` is a common reducer, why not just use 1 action, then specify the language in the payload / params. Language specific logic can then be added in the reducers if required. 

A common action like that wouldn't be so useful if different languages require very different actions/reducers from each other. In those cases, it might be better to stick to what you have done. Have you thought about the kinds of actions that Java requires?

Again, I'm not too sure what is the best approach - thoughts?